noxfile.py

```diff
@@ -1,9 +1,47 @@
+from pathlib import Path
+
 import nox
 
-@nox.session(python=["3.11"]) # The python version should match the readthedocs configuration.
+requirements_files = list(
+    {path.name.replace(".in", "") for path in Path(".").glob("*in")}
+)
+
+
+@nox.session(name="pip-compile", python=["3.11"])
+@nox.parametrize(["req"], requirements_files, requirements_files)
+def pip_compile(session: nox.Session, req: str):
+    """Generate lock files from input files or upgrade packages in lock files."""
+    # fmt: off
+    session.install(
+      "-r", "pip-tools.in",
+      "-c", "pip-tools.txt",
+    )
+    # fmt: on
+
+    # Use --upgrade by default unless a user passes -P.
+    upgrade_related_cli_flags = "-P", "--upgrade-package", "--no-upgrade"
```

Comment:
I think this is more readable:

```suggestion
    upgrade_related_cli_flags = ("-P", "--upgrade-package", "--no-upgrade")
```

Reply:
Honestly, I think it's more readable without the unnecessary parens :man_shrugging:

Reply:
Looking at this suggestion again I think the parens fit nicely. My initial thought was if they weren't strictly necessary then maybe it's better to leave them out. On second look though I think the parens make it feel a bit more cohesive.